src/nnet3/nnet-am-decodable-simple.cc

```diff
@@ -174,24 +175,16 @@ void NnetDecodableBase::GetOutputForFrame(int32 subsampled_frame,
       subsampled_frame - current_log_post_subsampled_offset_));
 }
 
-void NnetDecodableBase::GetCurrentIvector(int32 output_t_start,
-                                          int32 num_output_frames,
-                                          Vector<BaseFloat> *ivector) {
+void NnetDecodableBase::GetCurrentIvector(int32 frame_to_search,
```

Comment:
Could you add comments in this function on what exactly is happening ?
e.g. if ivector_ is not Null
       then it means we are computing a single ivector for the entire utterance and this has been specified in the constructor so we will just return that
       elif online_ivector_feats_ is Null:
              then no ivectors have been specified either as online ivectors or as ivector for utterance, so we just return. (What if the input ivector reference was not properly initialized, what happens in that case ?)
      else:
           return one ivector from online_ivector_feats_